internal/operator/cluster/pmm.go

```diff
@@ -0,0 +1,69 @@
+package cluster
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/percona/percona-postgresql-operator/internal/kubeapi"
+	"github.com/percona/percona-postgresql-operator/internal/operator"
+	crv1 "github.com/percona/percona-postgresql-operator/pkg/apis/crunchydata.com/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+)
+
+func UpdatePMMSidecar(clientset kubeapi.Interface, cluster *crv1.Pgcluster, deployment *appsv1.Deployment) error {
+	// need to determine if we are adding or removing
+	if cluster.Spec.PMM.Enabled {
+		return AddPMMSidecar(cluster, cluster.Name, deployment)
+	}
+
+	RemovePMMSidecar(deployment)
+
+	return nil
+}
+
+func AddPMMSidecar(cluster *crv1.Pgcluster, name string, deployment *appsv1.Deployment) error {
+	// use the legacy template generation to make the appropriate substitutions,
+	// and then get said generation to be placed into an actual Container object
+	template := operator.GetPMMContainer(cluster, name)
+	container := v1.Container{}
+
+	if err := json.Unmarshal([]byte(template), &container); err != nil {
+		return fmt.Errorf("error unmarshalling exporter json into Container: %w ", err)
+	}
+
+	// append the container to the deployment container list. However, we are
+	// going to do this carefully, in case the pmm container already exists.
+	// this definition will supersede any exporter container already in the
+	// containers list
+	containers := []v1.Container{}
+	for _, c := range deployment.Spec.Template.Spec.Containers {
```

Comment:
Actually, you can call there `RemovePMMSidecar` and then just append it

Reply:
done in https://github.com/percona/percona-postgresql-operator/pull/16/commits/60618860653130cf1d2c5e1be71789775896b9ec